app/src/sharedTest/java/org/oppia/android/app/testing/SplashTestActivityTest.kt

```diff
@@ -0,0 +1,202 @@
+import org.oppia.android.testing.OppiaTestRule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG_VALUE
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [SplashTestActivity]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = SplashTestActivityTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class SplashTestActivityTest {
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var platformParameterController: PlatformParameterController
+
+  private val mockPlatformParameterList by lazy {
+    val mockSplashScreenWelcomeMsgParam = PlatformParameter.newBuilder()
+      .setName(SPLASH_SCREEN_WELCOME_MSG)
+      .setBoolean(SPLASH_SCREEN_WELCOME_MSG_VALUE)
+      .build()
+
+    listOf<PlatformParameter>(
+      mockSplashScreenWelcomeMsgParam
+    )
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+  }
+
+  @Test
+  fun testSplashTestActivity_readEmptyDatabase_checkWelcomeMsgIsInvisibleByDefault() {
+    launch(SplashTestActivity::class.java).use {
+      onView(withText(SplashTestActivity.WELCOME_MSG)).check(doesNotExist())
+    }
+  }
+
+  @Test
+  fun testSplashTestActivity_updateEmptyDatabase_readDatabaseValues_checkWelcomeMsgIsVisible() {
+    platformParameterController.updatePlatformParameterDatabase(mockPlatformParameterList)
+    testCoroutineDispatchers.runCurrent()
+
+    launch(SplashTestActivity::class.java).use {
+      onView(withText(SplashTestActivity.WELCOME_MSG))
+        .inRoot(getToastMatcher())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  private fun getToastMatcher(): TypeSafeMatcher<Root?> {
```

Comment:
Suggest adding a comment about what this helper is doing.

Reply:
Done

Reply:
This can be written as a regular // comment rather than a KDoc comment. Also, maybe `Returns a [TypeSafeMatcher] which checks for any [Toast] message in the screen.`

Since it's not checking for a specific toast, right?

Reply:
This can be written as a regular // comment rather than a KDoc comment. Also, maybe `Returns a [TypeSafeMatcher] which checks for any [Toast] message in the screen.`

Since it's not checking for a specific toast, right?

Reply:
Yeah, Done